src/device_info.sh

```diff
@@ -341,6 +344,45 @@ function get_desktop_environment()
   device_info_data['desktop_environment']="$formatted_de"
 }
 
+# This function populates the kernel_name and kernel_version values from the
+# device_info_data variable.
+function get_kernel()
+{
+  local target="$1"
+  local flag="$2"
+  local cmd_name
+  local kernel_name
+  local cmd_version
+  local kernel_version
+
+  cmd_name='uname -s'
+  cmd_version="uname -r"
+
+  case "$target" in
+    1) # VM_TARGET
+      kernel_name=$("$cmd_name")
+      kernel_version=$("$cmd_version")
+      ;;
+    2) # LOCAL_TARGET
+      show_verbose "$flag" "$cmd_name"
+      kernel_name=$(cmd_manager 'SILENT' "$cmd_name")
+
+      show_verbose "$flag" "$cmd_version"
+      kernel_version=$(cmd_manager 'SILENT' "$cmd_version")
+      ;;
+    3) # REMOTE_TARGET
+      show_verbose "$flag" "$cmd_name"
+      kernel_name=$(cmd_manager 'SILENT' "$cmd_name")
+
+      show_verbose "$flag" "$cmd_version"
+      kernel_version=$(cmd_remotely "$cmd_version" 'SILENT')
```

Comment:
How about concatenating all the `uname` commands for the remote case and processing the string? I prefer to reduce the `cmd_remotely` to just one call.